### PR TITLE
SQS Listeners

### PR DESCRIPTION
**Pull request by muthusubra**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by Mhowell494 — CHANGES_REQUESTED** [review]
Looks good, just a few minor suggestions :)

Only the dev values are populated, can we also add the credentials for the other environments too?

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/config/SqsProperties.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Data
+@Component
+@ConfigurationProperties(prefix = "cloud-platform.aws.sqs")
+public class SqsProperties {
+    private String region;
+    private String accesskey;
```
[the remark this answers is not in the document]
Comment (muthusubra) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/config/SqsProperties.java]: Done

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecutionconcluded/listener/ProsecutionConcludedListener.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+@XRayEnabled
```
[the remark this answers is not in the document]
Comment (muthusubra) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecutionconcluded/listener/ProsecutionConcludedListener.java]: Done

File: crown-court-proceeding/src/main/resources/application.yaml [hunk 3]
```diff
@@ -49,3 +49,13 @@ spring:
...
+
+cloud-platform:
+  aws:
+    sqs:
+      region: ${CLOUD_PLATFORM_QUEUE_REGION}
+      accesskey: ${CLOUD_PLATFORM_QUEUE_ACCESS_KEY}
+      secretkey: ${CLOUD_PLATFORM_QUEUE_SECRET_KEY}
+      queue:
+        hearingResulted: ${HEARING_RESULTED_QUEUE}
+        prosecutionConcluded: ${LAA_PROSECUTION_CONCLUDED_QUEUE}
```
[the remark this answers is not in the document]
Comment (muthusubra) [reply crown-court-proceeding/src/main/resources/application.yaml:61]: Done

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+@XRayEnabled
+public class HearingResultedListener {
+
+    @JmsListener(destination = "${HEARING_RESULTED_QUEUE}", concurrency = "1")
```
[the remark this answers is not in the document]
Comment (muthusubra) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java:13]: Done

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+@XRayEnabled
```
[the remark this answers is not in the document]
Comment (muthusubra) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java]: Done

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.laa.crime.crowncourt.hearing.listener;
+
+import com.amazonaws.xray.spring.aop.XRayEnabled;
+import com.google.gson.Gson;
```
[the remark this answers is not in the document]
Comment (muthusubra) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java]: Done

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecutionconcluded/listener/ProsecutionConcludedListener.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+public class ProsecutionConcludedListener {
+
+    private final Gson gson;
```
[the remark this answers is not in the document]
Comment (muthusubra) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecutionconcluded/listener/ProsecutionConcludedListener.java]: Done

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Mhowell494) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/hearing/listener/HearingResultedListener.java:13]: I actually think the approach that refers to the application.yaml is more appropriate, as the environment variables you are referencing might not always be present.

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/config/SqsClientConfig.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class SqsClientConfig {
+    private final SqsProperties sqsProperties;
+
+    public AmazonSQS awsSqsClient() {
+        return AmazonSQSClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(sqsProperties.getAccessKey(), sqsProperties.getSecretKey()))).withRegion(Regions.fromName(sqsProperties.getRegion())).build();
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/config/SqsClientConfig.java:17]: Can this be formatted so it's a bit more readable?

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecutionconcluded/listener/ProsecutionConcludedListener.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (Mhowell494) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecutionconcluded/listener/ProsecutionConcludedListener.java]: Still needs to remove the dependency in the build.gradle

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/config/SqsClientConfig.java (hunk 8 above)
Comment (muthusubra) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/config/SqsClientConfig.java:17]: This is intelliJ Auto-formatted. But, I'll manually format it now

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

**Review by venkat980 — APPROVED** [review]

**Pull request closed without merging**